WikiDBUpdaterHttp/WikiDBUpdaterHttp.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using Azure.Data.Tables;
+using WikiSentiment;
+
+
+namespace WikiDBUpdaterHttp
+{
+    /// <summary>
+    /// HTTP triggered function that updates DB with new wiki collections
+    /// </summary>
+    public class WikiDBUpdaterHttp
+    {
+        private ConfigurationWrapper config;
+
+        private string[] allLanguageCodes;
+
+        private HttpClient httpClient;
+
+        private Dictionary<string, string[]> articleExceptions;
+
+        public WikiDBUpdaterHttp(IConfiguration iConfig)
+        {
+            config = new ConfigurationWrapper(iConfig);
+
+            articleExceptions = config.GetValue<Dictionary<string, string[]>>
+                ("FunctionValues:CountryExceptions");
+
+            allLanguageCodes = config.GetValue<string[]>("FunctionValues:CountryCodes");
+
+
+            httpClient = new HttpClient();
+
+            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer",
+                 config.GetValue<string>("WikiKeys:WikiAPIToken"));
+
+            httpClient.DefaultRequestHeaders.Add("Api-User-Agent",
+                config.GetValue<string>("WikiKeys:WikiUserContact"));
+        }
+
+        /// <summary>
+        /// Azure binding that launches the function
+        /// </summary>
+        /// <param name="req"></param>
+        /// <param name="tableClient"></param>
+        /// <param name="log"></param>
+        /// <returns></returns>
+        [FunctionName("WikiDBUpdaterHttp")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            [Table("datatables"), StorageAccount("AzureWebJobsStorage")] TableClient tableClient, 
+            ILogger log)
+        {
+            //url schema is /WikiDBUpdaterHttp?date=2021-12-31&days=3&discard&language=de,fr
+            //get and validate options
+            string YYYYMMDD = req.Query["date"];
+            DateTime startDate;
+            if (YYYYMMDD == null)
+                return new BadRequestObjectResult("Missing parameter: date");
+            if (!validateDate(YYYYMMDD, out startDate))
+                return new BadRequestObjectResult("Bad starting date: " + YYYYMMDD);
+
+            int daysToGo = 1;
+            string daysToGoString = req.Query["days"];
+            if (daysToGoString != null)
+                if (!int.TryParse(daysToGoString, out daysToGo))
+                    return new BadRequestObjectResult("Bad days parameter: " + daysToGoString);
```

Comment:
Don't nest like this.
```suggestion
            if (daysToGoString == null)
                return new BadRequestObjectResult($"Missing parameter: {nameof(daysToGoString)}");
             
            if (!int.TryParse(daysToGoString, out daysToGo))
                return new BadRequestObjectResult("Bad days parameter: " + daysToGoString);
```
Here it isn't too bad, but in general, try to structure your code to have the least possible amount of nesting. It makes it easier for others/future you to follow the logic. Here it also has the advantage it provides more fine grained error messages (which is usually a positive thing outside security)